plugins/synaptics-cape/fu-synaptics-cape-device.c

```diff
@@ -28,7 +28,8 @@
 #define FU_SYNAPTICS_CAPE_WORD_IN_BYTES	      4	 /* bytes */
 
 #define FU_SYNAPTICS_CAPE_CMD_APP_ID(a, b, c, d)                                                   \
-	((((a)-0x20) << 8) | (((b)-0x20) << 14) | (((c)-0x20) << 20) | (((d)-0x20) << 26))
+	((guint32)((((a)-0x20) << 8) | (((b)-0x20) << 14) | (((c)-0x20) << 20) |                   \
+		   (((d)-0x20) << 26)))
```

Comment:
To be honest, this confused me for the first review. Maybe just include the final value as a 0x1234 integer?

Reply:
Do you mean something like `#define FUSYNAPTICS_CAPE_APP_ID_CTRL  0xaabbcc00 ` ?

Reply:
Yes, maybe. I think that looks a ton less scary than that macro :)

Reply:
Does the macro really need as many parenthesis as it has?  Maybe just taking some off would make it more appealing too.
```
(a-0x20) << 8 | \
(b-0x20) << 14 | \
(c-0x20) << 20 | \
(d-0x20) << 6  \
```

Isn't that bad if the order of operations works as I thought it should.

Reply:
My brain is trying to work out what happens if a,b,c,d are uint8_t's, or int32_t's

Reply:
Also - what happens if any of those values are less than `0x20`?  It seems like the macro might only be safe for values 0x20 or greater.  In which case maybe back to @hughsie's point it should be a function.  Maybe something like this:

```
static inline guint32 fu_synaptics_cape_app_id_ctrl (guint a, guint b, guint c, guint d)
{
       g_return_val_if_fail (a > 0x20, 0);
       g_return_val_if_fail (b > 0x20, 0);
       g_return_val_if_fail (c > 0x20, 0);
       g_return_val_if_fail (d > 0x20, 0);
       return (a-0x20) << 8 |  (b-0x20) << 14 |  (c-0x20) << 20 | (d-0x20) << 26;
}
```

Reply:
But we only ever do `FU_SYNAPTICS_CAPE_CMD_APP_ID('C', 'T', 'R', 'L')` -- just hardcode it.

Reply:
Your point that if they want to add a second command other than CTRL then introduce a function then?

Reply:
No, there is no other than CTRL for firmware update case.